azurecontainerapps/azurecontainerapps.ts

```diff
@@ -177,30 +190,43 @@ export class azurecontainerapps {
                 console.log(tl.loc('CreateImageWithBuilderMessage'));
 
                 // Set the Oryx++ Builder as the default builder locally
-                new ContainerAppHelper().setDefaultBuilder();
+                appHelper.setDefaultBuilder();
 
                 // Create a runnable application image
-                new ContainerAppHelper().createRunnableAppImage(imageToDeploy, appSourcePath, runtimeStack);
+                appHelper.createRunnableAppImage(imageToDeploy, appSourcePath, runtimeStack);
+
+                // If telemetry is enabled, log that the builder scenario was targeted for this task
+                telemetryHelper.setBuilderScenario();
             }
 
             // If a Dockerfile was found or provided, create a runnable application image from that
             if (!util.isNullOrEmpty(dockerfilePath) && shouldBuildAndPushImage) {
                 console.log(tl.loc('CreateImageWithDockerfileMessage', dockerfilePath));
-                new ContainerAppHelper().createRunnableAppImageFromDockerfile(imageToDeploy, appSourcePath, dockerfilePath);
+                appHelper.createRunnableAppImageFromDockerfile(imageToDeploy, appSourcePath, dockerfilePath);
+
+                // If telemetry is enabled, log that the Dockerfile scenario was targeted for this task
+                telemetryHelper.setDockerfileScenario();
             }
 
             // Push image to Azure Container Registry
             if (shouldBuildAndPushImage) {
                 new ContainerRegistryHelper().pushImageToAcr(imageToDeploy);
+            } else {
+                // If telemetry is enabled, log that the previously built image scenario was targeted for this task
+                telemetryHelper.setImageScenario();
             }
 
             // Create or update Azure Container App
-            new ContainerAppHelper().createOrUpdateContainerApp(containerAppName, resourceGroup, imageToDeploy, optionalCmdArgs);
+            appHelper.createOrUpdateContainerApp(containerAppName, resourceGroup, imageToDeploy, optionalCmdArgs);
+            telemetryHelper.setSuccessfulResult();
         } catch (err) {
             tl.setResult(tl.TaskResult.Failed, err.message);
         } finally {
             // Logout of Azure if logged in during this task session
             authHelper.logoutAzure();
+
+            // Log telemetry for this task run (will do nothing if telemetry was previously disabled)
+            telemetryHelper.log();
```

Comment:
Nit: please consider renaming `log` to `sendLog` or something showing that it's where the log is actually sent. Else, your comment is the only thing that made it clear that here we're not just logging an empty log or something. :)

Reply:
Ah good call -- let me change that.